src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.Unix.cs

```diff
@@ -6,11 +6,12 @@
 
 namespace System.Diagnostics
 {
-    public static partial class Debug
+    public partial class DebugProvider
     {
         private static readonly bool s_shouldWriteToStdErr = Environment.GetEnvironmentVariable("COMPlus_DebugWriteToStdErr") == "1";
 
-        private static void ShowDialog(string stackTrace, string message, string detailMessage, string errorSource)
+        [System.Diagnostics.Conditional("DEBUG")]
```

Comment:
Why do these methods have `[System.Diagnostics.Conditional("DEBUG")]` ? I think this attribute will cause this to not work when you compile release CoreLib.

Reply:
I think ShowDialog and Write can be ignored when compiled in release. 

Here are the usages:

- In Debug.Assert (and Fail), 
- In Debug.Write
- As DefaultTraceDebugProvider in [DefaultTraceListener](https://github.com/dotnet/corefx/pull/32846/files/c93589ae274d47c213c01fdd81766b1eb18ea323#diff-2d9baaeca242fec8bc2788a0641c0391)
- As TraceProvider in [TraceInternal](https://github.com/dotnet/corefx/pull/32846/files/c93589ae274d47c213c01fdd81766b1eb18ea323#diff-fb96fc7e09871393f2de2c96cd2b3b79)

Reply:
These methods cannot be ignored when compiling Release CoreLib. If they were ignored when compiling Release CoreLib, these Debug APIs would be no-ops in the shipping product - not very useful.